vllm/distributed/device_communicators/cuda_wrapper.py

```diff
@@ -51,6 +51,9 @@ def find_loaded_library(lib_name) -> Optional[str]:
         return None
     start = line.index("/")
     path = line[start:].strip()
+    filename = path.split("/")[-1]
+    assert filename.startswith(lib_name) and filename.endswith(".so"), \
+        f"Unexpected filename: {filename} for library {lib_name}"
```

Comment:
In my case, the shared object is `libcudart-blahblah.so.12`, so the library version should be truncated:

```suggestion
    filename = path.split("/")[-1]
    assert filename.rpartition(".so")[0].startswith(lib_name), \
        f"Unexpected filename: {filename} for library {lib_name}"
```

Reply:
With the above suggested change the p2p check passes and vLLM server runs normal, thanks for quick fix!

Reply:
nice catch!